test/test_env.py

```diff
@@ -1121,6 +1123,17 @@ def test_batch_unlocked_with_batch_size(device):
         env.step(td_expanded)
 
 
+@pytest.mark.skipif(not _has_gym, reason="no gym")
+def test_info_dict_reader(seed=0):
+    import gym
+
+    env = GymWrapper(gym.make("HalfCheetah-v4"))
+    env.set_info_dict_reader(default_info_dict_reader(["x_position"]))
+
+    assert "x_position" in env.observation_spec.keys()
+    assert isinstance(env.observation_spec["x_position"], UnboundedContinuousTensorSpec)
```

Comment:
is there something else I could be checking here?

Reply:
you could do `assert env.observation_spec["x_position"].is_in(tensordict["x_position"])`

Reply:
it's essentially testing that x_position is what it's supposed to be 
Another thing we could do is set up a screwed up tensor spec (e.g. one that has wrong shape or wrong dtype) and check that this assertion raises an error

Reply:
I just added some additional tests as you suggested, although it wasn't raising an error at any stage, just returning false for 

```python
env.observation_spec["x_position"].is_in(tensordict["x_position"])
```

Would you have expected an exception at some point?

Reply:
no but yes if you put an assert if front of it :p